aiohttp/tracing.py

```diff
@@ -342,8 +350,8 @@ class Trace:
     def __init__(
         self,
         session: "ClientSession",
-        trace_config: TraceConfig,
-        trace_config_ctx: SimpleNamespace,
+        trace_config: TraceConfig[object],
+        trace_config_ctx: Any,
```

Comment:
```suggestion
        trace_config: TraceConfig[_T],
        trace_config_ctx: _T,
```

Not sure if the TypeVar can be used here or not

Reply:
It would require making Trace a Generic, which I think starts to add overhead to people's code, particularly as there's no way to pass the type through automatically, so every tracing handler would need manual annotations. Not sure if we want to go down that route...

Reply:
This is essentially my cutoff point between type safety and keeping the code to allow anything the user wants.